lib/simplifyExpression/basicsSearch/removeAdditionOfZero.js

```diff
@@ -1,30 +1,32 @@
-const {rewriteNode} = require('math-rules');
+const clone = require('../../util/clone');
 
 const ChangeTypes = require('../../ChangeTypes');
-const defineRuleString = require('../../util/defineRuleString');
 const Node = require('../../node');
 
-// e.g. 2 + 0 -> 2
-const REMOVE_ADDING_ZERO = defineRuleString('#a + 0', '#a');
-
-// e.g. 0 + 2 -> 2
-const REMOVE_ADDING_ZERO_REVERSE = defineRuleString('0 + #a', '#a');
+const {query} = require('math-nodes');
 
 // If `node` is an addition node with 0 as one of its operands,
 // remove 0 from the operands list. Returns a Node.Status object.
 function removeAdditionOfZero(node) {
-  let newNode = null;
-  newNode = rewriteNode(REMOVE_ADDING_ZERO, node);
```

Comment:
I think there was a desire to introduce the use of `math-rules` as a separate step.